tests/integration/clientv3/metrics_test.go

```diff
@@ -168,11 +172,198 @@ func getHTTPBodyAsLines(t *testing.T, url string) []string {
+		"etcd_mvcc_hash_rev_duration_seconds_count",
+		"etcd_mvcc_hash_rev_duration_seconds_sum",
+		"etcd_mvcc_put_total",
+		"etcd_mvcc_range_total",
+		"etcd_mvcc_txn_total",
+		"etcd_network_client_grpc_received_bytes_total",
+		"etcd_network_client_grpc_sent_bytes_total",
+		"etcd_network_known_peers",
+		"etcd_server_apply_duration_seconds_bucket",
+		"etcd_server_apply_duration_seconds_count",
+		"etcd_server_apply_duration_seconds_sum",
+		"etcd_server_client_requests_total",
+		"etcd_server_go_version",
+		"etcd_server_has_leader",
+		"etcd_server_health_failures",
+		"etcd_server_health_success",
+		"etcd_server_heartbeat_send_failures_total",
+		"etcd_server_id",
+		"etcd_server_is_leader",
+		"etcd_server_is_learner",
+		"etcd_server_leader_changes_seen_total",
+		"etcd_server_learner_promote_successes",
+		"etcd_server_proposals_applied_total",
+		"etcd_server_proposals_committed_total",
+		"etcd_server_proposals_failed_total",
+		"etcd_server_proposals_pending",
+		"etcd_server_quota_backend_bytes",
+		"etcd_server_read_indexes_failed_total",
+		"etcd_server_slow_apply_total",
+		"etcd_server_slow_read_indexes_total",
+		"etcd_server_snapshot_apply_in_progress_total",
+		"etcd_server_version",
+		"etcd_snap_db_fsync_duration_seconds_bucket",
+		"etcd_snap_db_fsync_duration_seconds_count",
+		"etcd_snap_db_fsync_duration_seconds_sum",
+		"etcd_snap_db_save_total_duration_seconds_bucket",
+		"etcd_snap_db_save_total_duration_seconds_count",
+		"etcd_snap_db_save_total_duration_seconds_sum",
+		"etcd_snap_fsync_duration_seconds_bucket",
+		"etcd_snap_fsync_duration_seconds_count",
+		"etcd_snap_fsync_duration_seconds_sum",
+		"grpc_client_handled_total",
+		"grpc_client_msg_received_total",
+		"grpc_client_msg_sent_total",
+		"grpc_client_started_total",
+		"grpc_server_handled_total",
+		"grpc_server_handling_seconds_bucket",
+		"grpc_server_handling_seconds_count",
+		"grpc_server_handling_seconds_sum",
+		"grpc_server_msg_received_total",
+		"grpc_server_msg_sent_total",
+		"grpc_server_started_total",
+	}
+
+	// Get the list of generated metrics
+	generatedMetrics := getMetricsList(t, url)
+	for _, metric := range expectedMetrics {
+		if !contains(generatedMetrics, metric) {
+			t.Errorf("Expected metric %s not found in generated metrics", metric)
+		}
+	}
+}
+
+func getMetricsList(t *testing.T, url string) []string {
+	lines := getHTTPBodyAsLines(t, url)
+	metrics := make(map[string]struct{})
+	for _, line := range lines {
+		if strings.Contains(line, "{") {
+			metric := line[:strings.Index(line, "{")]
+			metrics[metric] = struct{}{}
+		} else {
+			metric := line[:strings.Index(line, " ")]
+			metrics[metric] = struct{}{}
+		}
+	}
+	var metricList []string
+	for metric := range metrics {
+		metricList = append(metricList, metric)
+	}
+	return metricList
+}
+
+func contains(slice []string, item string) bool {
+	for _, s := range slice {
+		if s == item {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
You should be able to use [slices.Contains](https://pkg.go.dev/slices#Contains) directly.

Reply:
Done